File: android/app/src/debug/java/com/laneshadow/sandbox/stories/organisms/LSInlineErrorCalloutStory.kt
package com.laneshadow.sandbox.stories.organisms

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.theme.LaneShadowTheme
import com.laneshadow.ui.organisms.LSInlineErrorCallout
import com.laneshadow.ui.organisms.SuggestionChip
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSInlineErrorCalloutStory {
    val all: List<Story> = listOf(
        Story(
            id = "organisms.inlineerror.error-only",
            tier = ComponentTier.Organism,
            component = "LSInlineErrorCallout",
            name = "Error Only",
            summary = "Warning compass chip + THE NAVIGATOR label + error body in Newsreader. No detail or suggestions.",
            content = { ErrorOnlyStory() },
        ),
        Story(
            id = "organisms.inlineerror.with-detail",
            tier = ComponentTier.Organism,
            component = "LSInlineErrorCallout",
            name = "With Detail",
            summary = "detail prop shown in ui.body.sm / content-secondary below the main body. Adds recoverable context.",
            content = { WithDetailStory() },
        ),
        Story(
            id = "organisms.inlineerror.with-suggestions",
            tier = ComponentTier.Organism,
            component = "LSInlineErrorCallout",
            name = "With Suggestions",
            summary = "Horizontal LSSuggestionChip row below dashed divider. Primary chip tinted warning amber. onSuggestionTap fires once.",
            content = { WithSuggestionsStory() },
        ),
        Story(
            id = "organisms.inlineerror.long-body-suggestions",
            tier = ComponentTier.Organism,
            component = "LSInlineErrorCallout",
            name = "Long Body + Long Suggestions",
            summary = "Long body + detail + three chips. Container expands vertically. Chips wrap if needed.",
            content = { LongBodySuggestionsStory() },
        ),
        Story(
            id = "organisms.inlineerror.dark-mode",
            tier = ComponentTier.Organism,
            component = "LSInlineErrorCallout",
            name = "Dark Mode",
            summary = "Warning amber border and chip tint remain legible on dark glass surface. Glass bg resolves to ink-tinted.",
            content = { DarkModeStory() },
        ),
    )
}

@Composable
private fun ErrorOnlyStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInlineErrorCallout(
                body = "Couldn't stitch that route — segment through Lucia looked broken.",
                detail = null,
                suggestions = emptyList(),
                onSuggestionTap = {},
            )
        }
    }
}

@Composable
private fun WithDetailStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInlineErrorCallout(
                body = "Couldn't stitch that one together.",
                detail = "The segment through Lucia looked broken. Try a different end point or route inland via Carmel Valley Rd instead.",
                suggestions = emptyList(),
                onSuggestionTap = {},
            )
        }
    }
}

@Composable
private fun WithSuggestionsStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInlineErrorCallout(
                body = "Couldn't stitch that one together — the segment through Lucia looked broken.",
                detail = "Try a different end point, or route inland.",
                suggestions = listOf(
                    SuggestionChip(label = "Try inland", isPrimary = true),
                    SuggestionChip(label = "End at Big Sur", isPrimary = false),
                ),
                onSuggestionTap = {},
            )
        }
    }
}

@Composable
private fun LongBodySuggestionsStory() {
    LaneShadowTheme {
        StoryColumn {
            LSInlineErrorCallout(
                body = "Couldn't stitch that one together — the segment through Lucia looked broken. The road data shows a closure between mile marker 40 and 44.",
                detail = "Try a different end point, or let me route you inland via Carmel Valley Rd instead? The views are different but the road is spectacular.",
                suggestions = listOf(
                    SuggestionChip(label = "Try inland via CVR", isPrimary = true),
                    SuggestionChip(label = "End at Big Sur", isPrimary = false),
                    SuggestionChip(label = "Avoid Lucia", isPrimary = false),
                ),
                onSuggestionTap = {},
            )
        }
    }
}

@Composable
private fun DarkModeStory() {
    LaneShadowTheme(darkTheme = true) {
        StoryColumn {
            LSInlineErrorCallout(
                body = "Couldn't stitch that one together.",
                detail = null,
                suggestions = listOf(
                    SuggestionChip(label = "Try inland", isPrimary = true),
                    SuggestionChip(label = "End at Big Sur", isPrimary = false),
                ),
                onSuggestionTap = {},
            )
        }
    }
}

@Composable
private fun StoryColumn(content: @Composable () -> Unit) {
    val theme = LocalLaneShadowTheme.current
    Column(
        modifier = Modifier.padding(theme.space.lg),
        verticalArrangement = Arrangement.spacedBy(theme.space.md),
    ) {
        content()
    }
}

File: ios/LaneShadow/Sandbox/Stories/Organisms/LSInlineErrorCalloutStory.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI

@MainActor
enum LSInlineErrorCalloutStory {
    static let all: [Story] = [
        Story(
            id: "organisms.inlineerror.error-only",
            tier: .organism,
            component: "InlineErrorCallout",
            name: "Error Only",
            summary: "Compass chip + 'THE NAVIGATOR' label + body in opinion.md. Warning stripe."
        ) { _ in
            LSInlineErrorCallout(
                body: "Couldn't stitch together a continuous route.",
                onSuggestionTap: { _ in }
            )
        },
        Story(
            id: "organisms.inlineerror.with-detail",
            tier: .organism,
            component: "InlineErrorCallout",
            name: "With Detail",
            summary: "Additional detail text in ui.body.sm with content.textMuted color."
        ) { _ in
            LSInlineErrorCallout(
                body: "Couldn't stitch together a continuous route.",
                detail: "Try a different destination or start point.",
                onSuggestionTap: { _ in }
            )
        },
        Story(
            id: "organisms.inlineerror.with-suggestions",
            tier: .organism,
            component: "InlineErrorCallout",
            name: "With Suggestions",
            summary: "Horizontal LSSuggestionChip row for user actions. Chips fire onSuggestionTap."
        ) { _ in
            LSInlineErrorCallout(
                body: "Couldn't stitch together a continuous route.",
                detail: "Try a different destination or start point.",
                suggestions: ["Try inland routes", "End at Big Sur"],
                onSuggestionTap: { _ in }
            )
        },
        Story(
            id: "organisms.inlineerror.long-body-suggestions",
            tier: .organism,
            component: "InlineErrorCallout",
            name: "Long Body + Long Suggestions",
            summary: "Extended text and multiple suggestion chips to test wrapping."
        ) { _ in
            LSInlineErrorCallout(
                body: "Couldn't stitch together a continuous route between your selected points. The terrain and road network in this region don't support a seamless cycling path.",
                detail: "Consider alternative routes or adjusting your start/end points.",
                suggestions: ["Try inland routes", "End at Big Sur", "Adjust start point", "See alternatives"],
                onSuggestionTap: { _ in }
            )
        },
        Story(
            id: "organisms.inlineerror.dark-mode",
            tier: .organism,
            component: "InlineErrorCallout",
            name: "Dark Mode",
            summary: "Warning stripe and compass re-resolve to dark surface. Content colors adapt automatically."
        ) { _ in
            LSInlineErrorCallout(
                body: "Couldn't stitch together a continuous route.",
                detail: "Try a different destination or start point.",
                suggestions: ["Try inland routes", "End at Big Sur"],
                onSuggestionTap: { _ in }
            )
            .preferredColorScheme(.dark)
        },
    ]
}
